### ISSUE TITLE
Invalid number of arguments in backward

### ISSUE THREAD

**Issue by fwilliams** [getkeops/keops#116]
Hi, thanks again for this amazing library!

I'm attempting to backprop through a `Sum_reduction` operation in pytorch, but I am getting a very bizare error that doesn't make sense to me. Here's a succinct version of what I'm doing

```python
def keops_kernel_mmv(X1, X2):
    # formula = '((X|Y) / (Norm2(X) * Norm2(Y))) * b'
    formula = '(X|Y) * b'
    variables = [f'X = Vi({X1.shape[1]})',  # 1st argument, indexed by i, dim D.
                 f'Y = Vj({X2.shape[1]})',  # 2nd argument, indexed by j, dim D.
                 f'b = Vj(1)']
    return Genred(formula, variables, reduction_op='Sum', axis=1)

# model has differentiable parameters and outputs a tensor of shape [n, d] where n is the number of points passed in
feat_x = model(x)  
feat_xp = model(xp)
Keval = keops_kernel_mmv(feat_xp, feat_x)
fx = Keval(feat_xp, feat_x, alpha)
loss = nn.functional.mse_loss(fx, y_eval_i)
loss.backward()
```

The call to `backward()` produces the error:
```
  File "/home/francis/bin/miniconda3/envs/neural-splines/lib/python3.8/site-packages/torch/tensor.py", line 221, in backward
    torch.autograd.backward(self, gradient, retain_graph, create_graph)
  File "/home/francis/bin/miniconda3/envs/neural-splines/lib/python3.8/site-packages/torch/autograd/__init__.py", line 130, in backward
    Variable._execution_engine.run_backward(
  File "/home/francis/bin/miniconda3/envs/neural-splines/lib/python3.8/site-packages/torch/autograd/function.py", line 89, in apply
    return self._forward_cls.backward(self, *args)  # type: ignore
  File "/home/francis/bin/miniconda3/envs/neural-splines/lib/python3.8/site-packages/pykeops/torch/operations.py", line 134, in backward
    grad = genconv(formula_g, aliases_g, backend, dtype, device_id, ranges, accuracy_flags, *args_g)
  File "/home/francis/bin/miniconda3/envs/neural-splines/lib/python3.8/site-packages/pykeops/torch/generic/generic_red.py", line 51, in forward
    out = myconv.genred_pytorch_out(tagCPUGPU, tag1D2D, tagHostDevice, device_id, ranges,
RuntimeError: [KeOps] Wrong number of args : is 4 but should be at least 5 in Sum_Reduction (with tagI=0) of :
Variables : x3 (dim=1), y2 (dim=1), y1 (dim=256), 
Formula = ((y2*x3)*y1)
```

Any ideas?

**Comment by jeanfeydy** [comment]
Hi @fwilliams ,

Thanks for your kind words and your help on the trigonometric functions!
I was not able to reproduce your issue, but wonder if this could be due to a mismatch between tensor shapes. In your example, Genred will return a tensor of shape `(N, 1)` and *not* `(N,)`: did you take this into account in your code?

For reference, the code below works for me on Google Colab:

```python
import torch
from pykeops.torch import Genred
from torch import nn

def keops_kernel_mmv(X1, X2):
    # formula = '((X|Y) / (Norm2(X) * Norm2(Y))) * b'
    formula = '(X|Y) * b'
    variables = [f'X = Vi({X1.shape[1]})',  # 1st argument, indexed by i, dim D.
                 f'Y = Vj({X2.shape[1]})',  # 2nd argument, indexed by j, dim D.
                 f'b = Vj(1)']
    return Genred(formula, variables, reduction_op='Sum', axis=1)

# model has differentiable parameters and outputs a tensor of shape [n, d] where n is the number of points passed in
feat_x = torch.randn(100, 2).cuda()
feat_x.requires_grad = True
feat_xp = torch.randn(100, 2).cuda()
alpha = torch.randn(100, 1).cuda()
y_eval_i = torch.randn(100, 1).cuda()  # N.B.: (100, 1) and not (100,)!

Keval = keops_kernel_mmv(feat_xp, feat_x)
fx = Keval(feat_xp, feat_x, alpha)
loss = nn.functional.mse_loss(fx, y_eval_i)
loss.backward()

print(fx.shape)
print(loss)
```
and returns:
```python
torch.Size([100, 1])
tensor(653.5854, device='cuda:0', grad_fn=<MseLossBackward>)
```

If you can send us a minimal (non-)working example, troubleshooting would be easier of course :-) 
Best regards,
Jean

**Comment by fwilliams** [comment]
Hi @jeanfeydy sorry for the late reply. I'm recovering from the CVPR deadline. It looks like the version of KeOps installed in my environment was pointing to my fork which is an outdated version of the master branch. I installed the version on PyPI and the problem went away. Is the version on PyPI up to date? If so then this is likely fixed, if not then maybe it's still present.

**Comment by jeanfeydy** [comment]
Hi @fwilliams,

No worries of course, I was in the exact same situation :-) 
We've just released a new v1.4.2 version on PyPi: could you maybe re-run your code to see if the problem has, indeed, been solved?

Best regards,
Jean

**Comment by jeanfeydy** [comment]
The problem appears to be solved: feel free to re-open if needed!
